packages/core-mobile/app/services/walletconnectv2/WalletConnectService.ts

```diff
@@ -280,19 +285,93 @@ class WalletConnectService implements WalletConnectServiceInterface {
     })
   }
 
+  updateSessionForNonEvmAccounts = async ({
+    session,
+    account,
+    isTestnet = false
+  }: {
+    session: SessionTypes.Struct
+    account: CorePrimaryAccount
+    isTestnet?: boolean
+  }): Promise<void> => {
+    const topic = session.topic
+    Logger.info(
+      `updating WC session '${session.peer.metadata.name}' with non evm chains.`
+    )
+
+    const namespaces: SessionTypes.Namespaces = {}
+
+    for (const key of Object.keys(session.namespaces)) {
+      const namespace = session.namespaces[key]
+
+      if (!namespace) continue
+
+      // for the matching namespace, we need to update both chain and account lists
+      // for the rest, we just leave as is
+      if (key === BlockchainNamespace.AVAX) {
+        namespace.chains = isTestnet
+          ? [
+              AvalancheCaip2ChainId.C_TESTNET,
+              AvalancheCaip2ChainId.P_TESTNET,
+              AvalancheCaip2ChainId.X_TESTNET
+            ]
+          : [
+              AvalancheCaip2ChainId.C,
+              AvalancheCaip2ChainId.P,
+              AvalancheCaip2ChainId.X
+            ]
+
```

Comment:
updateSession is more about adding additional chain/account actual. it's not about replacing/deleting.